http/external/src/main/java/org/wildfly/security/http/external/ExternalAuthenticationMechanism.java

```
@@ -0,0 +1,107 @@
+public class ExternalAuthenticationMechanism implements HttpServerAuthenticationMechanism {
+
+    private final CallbackHandler callbackHandler;
+
+    ExternalAuthenticationMechanism(CallbackHandler callbackHandler) {
+        this.callbackHandler = callbackHandler;
+    }
+
+    /**
+     * @see org.wildfly.security.http.HttpServerAuthenticationMechanism#getMechanismName()
+     */
+    @Override
+    public String getMechanismName() {
+        return EXTERNAL_NAME;
+    }
+
+    @Override
+    public void evaluateRequest(HttpServerRequest request) throws HttpAuthenticationException {
+
+        String remoteUser = request.getRemoteUser();
+        if (remoteUser == null) {
+            request.noAuthenticationInProgress();
+            return;
+        }
+
+        if (authorize(remoteUser)) {
+            succeed(request);
+        } else {
+            fail(request);
+        }
+
+    }
+
+    private boolean authorize(String username) throws HttpAuthenticationException {
+        AuthorizeCallback authorizeCallback = new AuthorizeCallback(username, username);
+        try {
+            MechanismUtil.handleCallbacks(httpExternal, callbackHandler, authorizeCallback);
+            return authorizeCallback.isAuthorized();
+        } catch (AuthenticationMechanismException e) {
+            throw e.toHttpAuthenticationException();
+        } catch (UnsupportedCallbackException e) {
+            throw httpExternal.mechCallbackHandlerFailedForUnknownReason(e).toHttpAuthenticationException();
+        }
+    }
+
+    private void succeed(HttpServerRequest request) throws HttpAuthenticationException {
+        try {
+            MechanismUtil.handleCallbacks(httpExternal, callbackHandler, AuthenticationCompleteCallback.SUCCEEDED);
+            request.authenticationComplete();
+        } catch (AuthenticationMechanismException e) {
+            throw e.toHttpAuthenticationException();
+        } catch (UnsupportedCallbackException ignored) {
+            // ignored
```

Instead of ignoring this, I think we should throw an `HttpAuthenticationException` here.

---

Fixed both, Thanks!